lib/acts_as_taggable_on_mongoid/taggable/tagger_relation.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module ActsAsTaggableOnMongoid
+  module Taggable
+    # Overides of methods from Mongoid::Processing which process attributes for methods like
+    # `assign_attributes`, `create`, `new`, and `update_attributes`
+    #
+    # The need for this override is because the base method splits the order that methods are
+    # processed in to process relationships after processing other attributes.
+    #
+    # However, tag lists may rely upon values set in relationships - forcing us to process
+    # tag lists AFTER relationships have been processed - preventing the need to order attributes
+    # (which wouldn't help anyway because of the way process_attributes works.)
+    #
+    # ONLY taggings that have a default and which accept tagger values and which default the
+    # tagger based on the taggable object could be affected by other attributes set at the same time
+    # as the tag list.  Thus only those tag values are delayed until after all other attributes are set.
+    module TaggerRelation
+      def process_attributes(attrs = nil)
```

Comment:
I didn't want to override this method, but I couldn't find another way to do this.

The method appears to be a public method, so overriding it doesn't appear to be a major no-no.